packages/account-tree-controller/src/AccountTreeController.ts

```diff
@@ -394,44 +400,90 @@ export class AccountTreeController extends BaseController<
     wallet: AccountWalletObject,
     group: AccountGroupObject,
   ) {
-    const persistedMetadata = this.state.accountGroupsMetadata[group.id];
+    const persistedGroupMetadata = this.state.accountGroupsMetadata[group.id];
 
     // Apply persisted name if available (including empty strings)
-    if (persistedMetadata?.name !== undefined) {
-      group.metadata.name = persistedMetadata.name.value;
+    if (persistedGroupMetadata?.name !== undefined) {
+      group.metadata.name = persistedGroupMetadata.name.value;
     } else if (!group.metadata.name) {
       // Get the appropriate rule for this wallet type
       const rule = this.#getRuleForWallet(wallet);
       const typedWallet = wallet as AccountWalletObjectOf<typeof wallet.type>;
-      const typedGroup = typedWallet.groups[group.id] as AccountGroupObject;
-
-      // Calculate group index based on position within sorted group IDs
-      // We sort to ensure consistent ordering across all wallet types:
-      // - Entropy: group IDs like "entropy:abc/0", "entropy:abc/1" sort to logical order
-      // - Snap/Keyring: group IDs like "keyring:ledger/0xABC" get consistent alphabetical order
-      const sortedGroupIds = Object.keys(wallet.groups).sort();
-      let groupIndex = sortedGroupIds.indexOf(group.id);
-
-      // Defensive fallback: if group.id is not found in sortedGroupIds (should never happen
-      // in normal operation since we iterate over wallet.groups), use index 0 to prevent
-      // passing -1 to getDefaultAccountGroupName which would result in "Account 0"
-      /* istanbul ignore next */
-      if (groupIndex === -1) {
-        groupIndex = 0;
-      }
+      const typedGroup = typedWallet.groups[group.id] as AccountGroupObjectOf<
+        typeof group.type
+      >;
 
       // Use computed name first, then fallback to default naming if empty
-      group.metadata.name =
-        rule.getComputedAccountGroupName(typedGroup) ||
-        rule.getDefaultAccountGroupName(groupIndex);
+      const computedName = rule.getComputedAccountGroupName(typedGroup);
+
+      if (computedName) {
+        group.metadata.name = computedName;
+      } else {
+        // Generate default name and ensure it's unique within the wallet
+        let proposedName = '';
+        let groupIndex: number;
```

Comment:
I think `groupIndex` is misleading here. I would rename it to `proposedNameIndex` maybe?